core/src/main/scala/quasar/destination/gbq/GBQDestination.scala

```diff
@@ -0,0 +1,213 @@
+import quasar.api.resource.ResourceName
+import quasar.api.table.{ColumnType, TableColumn}
+
+import argonaut._, Argonaut._
+
+import cats.ApplicativeError
+import cats.effect.{Concurrent, ContextShift, Resource, Sync}
+import cats.implicits._
+
+import com.google.auth.oauth2.AccessToken
+
+import eu.timepit.refined.auto._
+
+import fs2.Stream
+
+import org.http4s.{
+  AuthScheme,
+  Credentials,
+  EntityEncoder,
+  MediaType,
+  Method,
+  Request,
+  Status,
+  Uri}
+import org.http4s.argonaut.jsonEncoderOf
+import org.http4s.client.Client
+import org.http4s.headers.{Authorization, `Content-Type`, Location}
+
+import org.slf4s.Logging
+
+import scalaz.NonEmptyList
+
+import shims._
+
+final class GBQDestination[F[_]: Concurrent: ContextShift: MonadResourceErr](
+    client: Client[F], 
+    config: GBQConfig,
+    sanitizedConfig: Json) extends Destination[F] with Logging {
+
+  def destinationType: DestinationType = DestinationType("gbq", 1L)
+
+  def sinks: NonEmptyList[ResultSink[F]] = NonEmptyList(gbqSink)
+
+  private def gbqSink: ResultSink[F] = 
+    ResultSink.csv[F](RenderConfig.Csv()) { (path, columns, bytes) =>
+      val gbqSchema = tblColumnToGBQSchema(columns)
+      
+      val tableNameF = path.uncons match {
+        case Some((ResourceName(name), _)) => name.pure[F]
+        case _ => MonadResourceErr[F].raiseError[String](
+          ResourceError.notAResource(path))
+      }
+
+      for {
+        tableName <- Stream.eval[F, String](tableNameF)
+        gbqJobConfig = formGBQJobConfig(gbqSchema, config, tableName)
+        accessToken <- Stream.eval(GBQAccessToken.token(config.authCfg.getBytes("UTF-8")))
+        _ <- Stream.eval((mkDataset(client, accessToken, config)))
+        eitherloc <- Stream.eval(mkGbqJob(client, accessToken, gbqJobConfig))
+        _ <- Stream.eval(
+          Sync[F].delay(
+            log.info(s"(re)creating ${config.project}.${config.datasetId}.${tableName} with schema ${columns.show}")))
+        _ <- eitherloc match {
+            case Right(locationUri) => {
+              upload(client, bytes, locationUri, accessToken)
+            }
+            case Left(e) =>
+              Stream.eval(Sync[F].delay(ApplicativeError[F, Throwable].raiseError(
+                new RuntimeException(s"No Location URL from returned from job: $e"))))
+          }
+      } yield ()
+  }
+
+  // NB: config overwrites when re-pushing tables
+  // see schemaUpdateOptions section at
+  // https://cloud.google.com/bigquery/docs/reference/rest/v2/Job
+  private def formGBQJobConfig(
+      schema: List[GBQSchema],
+      config: GBQConfig,
+      tableId: String): GBQJobConfig =
+    GBQJobConfig(
+      "CSV",
+      "1",
+      "true",
+      schema,
+      Some("DAY"),
+      WriteDisposition("WRITE_TRUNCATE"),
+      GBQDestinationTable(config.project, config.datasetId, tableId))
+
+  private def tblColumnToGBQSchema(cols: List[TableColumn]): List[GBQSchema] =
+    cols map { col => col match {
```

Comment:
⛳️ 

You don't need the `col => col match {` bit. Just literally delete it and remove the second trailing curly brace and all of this will work.

Reply:
⛳️